src/toolchain.rs

```diff
@@ -119,4 +175,35 @@ impl Toolchain {
 
         Ok(download_cfg)
     }
+
+    pub fn remove_component(&self, component: &str) -> Result<()> {
+        if self.has_component(component) {
+            info!("Removing '{}' from toolchain '{}'", component, self.name);
+            let component_path = self.path.join(component);
+            remove_file(component_path)?;
```

Comment:
Maybe we should provide some context here as I/O errors can be a bit ambiguous on their own, e.g. something like:
```rust
remove_file(component_path)
    .with_context(|| format!("failed to remove component '{}', component))?;
```